man/usleep.1
.TH USLEEP 1 "Red Hat, Inc" \" -*- nroff -*-
.SH NAME
usleep \- sleep some number of microseconds
.SH SYNOPSIS
.B usleep
[\fInumber\fP]
.SH DESCRIPTION
.B usleep
sleeps some number of microseconds.  The default is 1.
.SH WARNING
.B usleep
has been deprecated, and will be removed in near future. Use sleep(1) instead.
.SH OPTIONS
\fI--usage\fP
Show short usage message.
.TP
\fI--help, -?\fP
Print help information.
.TP
\fI-v, --version\fP
Print version information.
.SH BUGS
Probably not accurate on many machines down to the microsecond.  Count
on precision only to -4 or maybe -5.
.SH AUTHOR
Donald Barnes <djb@redhat.com>
.br
Erik Troan <ewt@redhat.com>
.SH SEE ALSO
sleep(1)


ChangeLog
commit f8c758709c2dea6651afcc75fa7cceb3b7ff7afc
Author: Harald Hoyer <harald@redhat.com>
Date:   Mon Sep 13 18:17:22 2010 +0200

    version 8.45.32

 initscripts.spec |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

commit 967b64ce3f9754f7de902ed290c813634c182803
Author: Harald Hoyer <harald@redhat.com>
Date:   Mon Sep 13 18:16:21 2010 +0200

    52a7a1f115bb87323620b73f5dde21e2c5e7d9a7

 ChangeLog                        |    7 ++++---
 udev/rules.d/54-cu3088-fix.rules |   10 ++++++++++
 2 files changed, 14 insertions(+), 3 deletions(-)

commit c014f9c6c09b50302e89f9df21077343b6c363b8
Author: Harald Hoyer <harald@redhat.com>
Date:   Mon Sep 13 17:34:33 2010 +0200

    initrscripts.spec: sync with cvs repo

 initscripts.spec |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

commit 923ef28cc4d4654e58033c81cd44ce9e1e4734b8
Author: Harald Hoyer <harald@redhat.com>
Date:   Mon Sep 13 17:30:23 2010 +0200

    version 8.45.31

 ChangeLog        | 1016 ++++++++++++++++++++++++++++++++++--------------------
 initscripts.spec |    4 +-
 2 files changed, 636 insertions(+), 384 deletions(-)

commit 52a7a1f115bb87323620b73f5dde21e2c5e7d9a7
Author: Phil Knirsch <pknirsch@redhat.com>
Date:   Mon Sep 13 16:29:45 2010 +0200

    s390: Added required 54-cu3088-fix.rules udev rules file (#475721)

 initscripts.spec |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

commit b74c943aa01f6a8facc36246dc20ccebd9eb690c
Author: David Cantrell <dcantrell@redhat.com>
Date:   Fri Sep 26 16:41:21 2008 -1000

    Expand get_hostname() function to support IPv4 and IPv6.
    
    Make sure that get_hostname() if the user specifies either address
    family.

 src/ipcalc.c |   36 +++++++++++++++++++++++++++++-------
 1 files changed, 29 insertions(+), 7 deletions(-)

commit f085cdfdc0d67fe240c23e75d14dce6534d36c8a
Author: David Cantrell <dcantrell@redhat.com>
Date:   Fri Sep 26 16:30:38 2008 -1000

    Remove unnecessary tab.

 src/ipcalc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

commit e921ac88353159dee451f7c0094fca95a9b2bb56
Author: David Cantrell <dcantrell@redhat.com>
Date:   Fri Sep 26 16:29:21 2008 -1000

    Correct address family option handling in ipcalc.
    
    Make sure the default is IPv4.  If the check option is specified,
    make sure we just validate the address and return 1 or 0.

 src/ipcalc.c |   83 ++++++++++++++++++++++++++++-----------------------------
 1 files changed, 41 insertions(+), 42 deletions(-)

commit 9253ed4bdddd635706a0a741259cbaa8a21d8b89
Author: David Cantrell <dcantrell@redhat.com>
Date:   Fri Sep 26 16:26:27 2008 -1000

    Compile ipcalc with _GNU_SOURCE defined (for asprintf).

 src/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

commit 192392a47594d8141aaf14a0a705bc421899adb0
Author: Harald Hoyer <harald@redhat.com>
Date:   Mon Sep 13 16:20:36 2010 +0200

    remove deprecation comments for raw devices and fix path

 initscripts.spec       |    6 ++++++
 rc.d/init.d/rawdevices |    6 ++----
 sysconfig/rawdevices   |    3 ---
 3 files changed, 8 insertions(+), 7 deletions(-)

commit 6c0c324a6c0c031ed4604c6a020829503e8d086f
Author: Harald Hoyer <harald@redhat.com>
Date:   Mon Sep 13 15:59:40 2010 +0200

    version 8.45.30

 initscripts.spec |   28 +++++++++++++++++++++++++++-
 1 files changed, 27 insertions(+), 1 deletions(-)

commit 8f65b61fc53d245c7865bfa0ab879337d9c6ff31
Author: Harald Hoyer <harald@redhat.com>
Date:   Mon Sep 13 15:48:34 2010 +0200

    rc.sysinit: also remove /var/run/dovecot/*
    
    https://bugzilla.redhat.com/show_bug.cgi?id=492994

 rc.d/rc.sysinit |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

commit 85f809ad1d6f47d4fe73a07fa358fd917bc94b87
Author: Bill Nottingham <notting@redhat.com>
Date:   Wed Sep 2 16:28:02 2009 -0400

    Doucment DHCP_HOSTNAME (#498052)

 sysconfig.txt |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

commit 52dfb70cd860e4d0fd51e37e30a081fd197f1b74
Author: Bill Nottingham <notting@redhat.com>
Date:   Fri Jul 20 20:54:47 2007 +0000

    move quotacheck to after filesystem mounting so it actually works (#249003, <tometzky@batory.org.pl>)
    also, nuke the convertquota stuff that has been obsolete (and not working) for quite some time

 rc.d/rc.sysinit |   25 +++++--------------------
 1 files changed, 5 insertions(+), 20 deletions(-)

commit 873660981a5a42a6de411e04958a4f204704efe1
Author: Bill Nottingham <notting@redhat.com>
Date:   Tue Dec 15 10:59:44 2009 -0500

    Silence error. (#516569)

 sysconfig/network-scripts/network-functions |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

commit 3ae2380e43092dfdc12b2d242ade6bd1379fe1e7
Author: Bill Nottingham <notting@redhat.com>
Date:   Tue Oct 20 16:56:32 2009 -0400

    Load bonding driver if BONDING_OPTS is defined. (#516569)

 sysconfig/network-scripts/network-functions |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

commit f08000519addd0b13fada0eba4afef72871613c1
Author: Bill Nottingham <notting@redhat.com>
Date:   Thu Aug 27 13:25:41 2009 -0400

    Restore context of recreated files. (#519748)

 rc.d/rc.sysinit |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

commit 209e5e18cbd1cbb76f7e49c83854763c4e9f1d1c
Author: Bill Nottingham <notting@redhat.com>
Date:   Fri Oct 9 16:18:04 2009 -0400

    Do not try to activate subsets and layer raid and handle 'no raid disks'.
    
    - 'no raid disks' handling: (#527726, <mschmidt@redhat.com>)
    - no subsets: https://bugzilla.redhat.com/show_bug.cgi?id=524356

 rc.d/rc.sysinit |   15 +++++++--------
 1 files changed, 7 insertions(+), 8 deletions(-)

commit 40ddde4c4e6ad1dc855a8a3f4ae594d6c2d16179
Author: Bill Nottingham <notting@redhat.com>
Date:   Wed Oct 7 11:10:12 2009 -0400

    Add /var/lib/xend. (#526046)

 rwtab |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

commit 643c05c59a581663140e307f08732c6bef3d7622
Author: Bill Nottingham <notting@redhat.com>

src/usleep.c
/* 
 * usleep
 * 
 * Written by Donald Barnes <djb@redhat.com> for Red Hat, Inc.
 * 
 * Copyright (c) 1997-2003 Red Hat, Inc. All rights reserved.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License, version 2,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 */


#include <unistd.h>
#include <stdlib.h>
#include <string.h>
#include <stdio.h>

#include "popt.h"

int main(const int argc, const char **argv) {
  unsigned long count;
  poptContext optCon;
  int showVersion = 0;
  int showOot = 0;
  int rc;
  const char * countStr = NULL;
  struct poptOption options[] = {
            { "version", 'v', POPT_ARG_NONE, &showVersion, 0, 
			"Display the version of this program, and exit" },
            { "oot", 'o', POPT_ARG_NONE, &showOot, 0, 
			"oot says hey!" },
	    POPT_AUTOHELP
            { 0, 0, 0, 0, 0 }
        };

  optCon = poptGetContext("usleep", argc, argv, options,0);
  /*poptReadDefaultConfig(optCon, 1);*/
  poptSetOtherOptionHelp(optCon, "[microseconds]");

  if ((rc = poptGetNextOpt(optCon)) < -1) {
	fprintf(stderr, "usleep: bad argument %s: %s\n", 
		poptBadOption(optCon, POPT_BADOPTION_NOALIAS), 
		poptStrerror(rc));
	return 2;
  }

  if (showVersion) {
      printf("usleep version 1.2\n	usleep --help for more info\n");
      return 0;
  }

  if (showOot) {
      printf("oot says hey!\n");
      return 0;
  }

  countStr = poptGetArg(optCon);

  if (countStr == NULL) count = 1;

  else if (countStr && poptGetArg(optCon)) {
      fprintf(stderr, "%s: exactly one argument (number of microseconds) "
      		"must be used\n", argv[0]);
      return 2;
  }

  else count = strtoul(countStr, NULL, 0); 

  fprintf(stderr, "warning: usleep is deprecated, and will be removed in near future!\n"
	          "warning: use \"sleep %.7g\" instead...\n", count / 1e6);

  usleep(count);
  return 0;
} 
